### PR TITLE
Add `implements` and `signature` to syntax highlighting

### PR DESCRIPTION
**Pull request by dbartol**
This PR adds syntax highlighting for the new, still-experimental "parameterized modules" feature in the QL language.

`signature` is added as an "annotation keyword", highlighted only if it appears before a module, class, or predicate declaration. `implements` is added as a context-sensitive keyword, available only within a module defintion (but outside the module body).

### CODE REVIEW

File: extensions/ql-vscode/syntaxes/ql.tmLanguage.yml [hunk 1]
```diff
@@ -668,6 +679,19 @@ repository:
     - match: '(?#simple-id)'
       name: entity.name.type.namespace.ql
 
+  # The `extends` clause of a class definition.
```
Comment (aeisenberg) [inline extensions/ql-vscode/syntaxes/ql.tmLanguage.yml:682]:
```suggestion
  # The `implements` clause of a class definition.
```

**Review by aeisenberg — COMMENTED** [review]
Hmmm...`signature` is working as expected, but `implements` is not. Is something wrong with this syntax?


_Extension_Development_Host__definitions_ql_—_vscode-codeql-starter__Workspace_●

File: extensions/ql-vscode/syntaxes/ql.tmLanguage.yml (hunk 1 above)
Comment (dbartol) [reply extensions/ql-vscode/syntaxes/ql.tmLanguage.yml:682]: Doesn't `implements` only apply to modules?

Comment (dbartol) [reply extensions/ql-vscode/syntaxes/ql.tmLanguage.yml:682]: (I admit that would have been more obvious had I remembered to change the comment...)

**Review by aeisenberg — APPROVED** [review]
Ah...yes. That works now. Thanks.

**Pull request merged**